scripts/triggers_check.sh

```diff
@@ -14,31 +14,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -eu
+# Make sure that there is a trigger for each existing solution
+set -eu pipefail
 
 shopt -s nullglob
 
+missing_var=""
 # Ensure all required env vars are supplied.
 for var in DIRECTORY_NAME CLOUDBUILD_NAME PROJECT; do
   if ! [[ -v "${var}" ]]; then
     echo "${var} env variable is required"
-    exit 1
+    missing_var=true
   fi
 done
 
+if [[ -n "${missing_var}" ]]; then
+  exit 1
+fi
+
 function trigger_exist {
   local -r solution=$1
 
-  gcloud alpha builds triggers list --project="${PROJECT}" --format json \
-    | jq -e --arg filename "${CLOUDBUILD_NAME}" --arg solution "${solution}" \
-    '.[] | select(.filename == $filename) | select(.substitutions._SOLUTION_NAME == $solution)'
-    
-   return $?
+  echo "${triggers}" | jq -e --arg filename "${CLOUDBUILD_NAME}" --arg solution "${solution}" \
+      '.[] | select(.filename == $filename) | select(.substitutions._SOLUTION_NAME == $solution)'
+  return $?
 }
 
 function main {
   local -i failure_cnt=0
 
+  triggers="$(gcloud alpha builds triggers list --project="${PROJECT}" --format json)"
```

Comment:
Please consider to provide it as argument to the `trigger_exist` function instead, and mark it as local variable.

local -r triggers=...

Reply:
Probably it also make sense to move this variable outside the main function, so it became a real global one.

Reply:
Please also note that Cloud Build runs all steps parallel.

Reply:
Definitely prefer passing two parameters to `trigger_exist` instead of relying on a global variable. Consider renaming the function to `triggers_include_solution` in this case.